### DISCUSSION TITLE
Update stuck

### DISCUSSION THREAD

**Discussion by matthew73210** [mcguirepr89/BirdNET-Pi#747] [General]
Not sure if anyone is having this issue, but when ever I click on update. Stuff happens but it stays on the number it was before. Heres the output after clicking on update 

+ sudo -u matthew git -C /home/matthew/BirdNET-Pi reset --hard
HEAD is now at d257b14 Making recordings actually usable for huge amounts of detections
+ set +x
+ sudo -u matthew git -C /home/matthew/BirdNET-Pi fetch origin main
From https://github.com/mcguirepr89/BirdNET-Pi
 * branch            main       -> FETCH_HEAD
+ set +x
+ sudo -u matthew git -C /home/matthew/BirdNET-Pi switch -C main --track origin/main
error: The following untracked working tree files would be overwritten by checkout:
	scripts/print_diagnostic_info.sh
Please move or remove them before you switch branches.
Aborting
+ set +x
+ my_dir=/home/matthew/BirdNET-Pi/scripts
+ sudo systemctl stop birdnet_server.service
+ sudo systemctl stop birdnet_recording.service
+ services=(chart_viewer.service spectrogram_viewer.service icecast2.service extraction.service birdnet_recording.service birdnet_log.service)
+ for i in "${services[@]}"
+ sudo systemctl restart chart_viewer.service
+ for i in "${services[@]}"
+ sudo systemctl restart spectrogram_viewer.service
+ for i in "${services[@]}"
+ sudo systemctl restart icecast2.service
+ for i in "${services[@]}"
+ sudo systemctl restart extraction.service
+ for i in "${services[@]}"
+ sudo systemctl restart birdnet_recording.service
+ for i in "${services[@]}"
+ sudo systemctl restart birdnet_log.service
+ sudo systemctl start birdnet_server.service
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ status=1
+ ((  status != 0  ))
+ logger '[/usr/local/bin/restart_services.sh] Unable to start birdnet_server.service... Looping until it start properly'
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sudo systemctl restart birdnet_server.service
+ sleep 45
+ grep 5050 /dev/fd/63
++ netstat -tulpn
tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      5669/python3        
+ sudo systemctl restart birdnet_analysis.service

Looks like I'm on the wrong branch ?
Weridly some of my config has disappeared too, like my apprise config.

**Comment by lloydbayley** [comment]
You appear to be on the correct branch but not the correct HEAD number.
I just updated mine and it shows HEAD is now at a439214 Revert "restart services if analysis hangs"
It seems it's about 15 commits behind which is weird.
Do you get any other error messages in the logs? The losing config is a bit of a concern.
What is in the birdnet.conf file in the way of APPRISE_* tags or are they missing altogether?

**Reply by krummrey** [reply]
Mine has been like that forever. Have to restart the machine to get it out of that loop.
`HEAD is now at 75bc991 issue #742`

**Reply by matthew73210** [reply]
Another weird thing is when ever I login via ssh it says that uptime is 1 min.
Going back to apprise, all the config lines are there just empty now. And 2 of them are right at the end of the file. Might be related to when I had issues changing model.
journelctl -f doesn't bring up anything out of the ordinary. I'm on a rtsp stream and it tries to connect every five seconds, even while connected. Causing the livestream.service to fail. Not sure where the other logs are kept though.

**Reply by matthew73210** [reply]
Ah it seems to be rebooting itself after x mins.

**Reply by ehpersonal38** [reply]
Let's first try to get you properly updated before we tackle any other issues - 

```
cd ~/BirdNET-Pi
git reset --hard
git checkout -f main
git clean  -d  -f .
git pull
bash scripts/update_birdnet.sh
```

**Reply by matthew73210** [reply]
+ sudo -u matthew git -C /home/matthew/BirdNET-Pi reset --hard
HEAD is now at 75bc991 issue #742
+ set +x
+ sudo -u matthew git -C /home/matthew/BirdNET-Pi fetch origin main
From https://github.com/mcguirepr89/BirdNET-Pi
 * branch            main       -> FETCH_HEAD
+ set +x
+ sudo -u matthew git -C /home/matthew/BirdNET-Pi switch -C main --track origin/main
Reset branch 'main'
Branch 'main' set up to track remote branch 'main' from 'origin'.
Your branch is up to date with 'origin/main'.
+ set +x
WARNING: The directory '/home/matthew/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.

[notice] A new release of pip available: 22.2.1 -> 23.0.1
[notice] To update, run: python3 -m pip install --upgrade pip
+ my_dir=/home/matthew/BirdNET-Pi/scripts
+ sudo systemctl stop birdnet_server.service
+ sudo systemctl stop birdnet_recording.service
+ services=(chart_viewer.service spectrogram_viewer.service icecast2.service extraction.service birdnet_recording.service birdnet_log.service)
+ for i in "${services[@]}"
+ sudo systemctl restart chart_viewer.service
+ for i in "${services[@]}"
+ sudo systemctl restart spectrogram_viewer.service
+ for i in "${services[@]}"
+ sudo systemctl restart icecast2.service
+ for i in "${services[@]}"
+ sudo systemctl restart extraction.service
+ for i in "${services[@]}"
+ sudo systemctl restart birdnet_recording.service
+ for i in "${services[@]}"
+ sudo systemctl restart birdnet_log.service
+ sudo systemctl start birdnet_server.service
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ for i in {1..5}
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sleep 5
+ systemctl is-active --quiet birdnet_server.service
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ status=1
+ ((  status != 0  ))
+ logger '[/usr/local/bin/restart_services.sh] Unable to start birdnet_server.service... Looping until it start properly'
+ grep 5050 /dev/fd/63
++ netstat -tulpn
+ sudo systemctl restart birdnet_server.service
+ sleep 45
+ grep 5050 /dev/fd/63
++ netstat -tulpn
tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      2801/python3        
+ sudo systemctl restart birdnet_analysis.service

Output seems more positive. Still got 19 above the update button though.
